### ISSUE TITLE
Build fails with -Werror=strict-aliasing

### ISSUE THREAD

**Issue by orlitzky** [Singular/Singular#1212]
We have a CI system that tests packages with `-Werror=strict-aliasing` on Gentoo because the warning indicates a problem that can affect link-time optimization (LTO). With the latest p16 we're seeing a build failure in Singular. The original bug report is https://bugs.gentoo.org/927675, but you should be able to reproduce it with `CXXFLAGS="-Werror=strict-aliasing"`:

```
countedref.cc: In member function 'LeftvShallow CountedRefData::operator*() const':
countedref.cc:111:91: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  111 | nst { return (broken()? LeftvShallow(): (const LeftvShallow&)m_data); }
      |                                                              ^~~~~~

```

**Comment by hannes14** [comment]
fixed with 83a4e40c59dd99ff3131cbdb57b24b456aa04b93

**Comment by orlitzky** [comment]
Works well, thank you!
